forge/test/operators/pytorch/nn/test_embedding.py

```diff
@@ -0,0 +1,422 @@
+
+        self.weight = torch.rand(
+            (self.kwargs["num_embeddings"], self.kwargs["embedding_dim"]), dtype=kwargs["weight_dtype"]
+        )
+        self.embedding = self.operator(**self.kwargs, _weight=self.weight)
+
+    def forward(self, x: torch.Tensor):
+        v1 = self.embedding(self.constant)
+        # v2 = torch.add(x, x)
+        v2 = self.embedding(x)
+        # add consume inputs
+        add = torch.add(v1, v2)
+        return add
+
+
+class TestVerification:
+
+    MODEL_TYPES = {
+        InputSource.FROM_ANOTHER_OP: ModelFromAnotherOp,
+        InputSource.FROM_HOST: ModelDirect,
+        InputSource.FROM_DRAM_QUEUE: ModelDirect,
+        InputSource.CONST_EVAL_PASS: ModelConstEvalPass,
+    }
+
+    @classmethod
+    def verify(
+        cls,
+        test_device: TestDevice,
+        test_vector: TestVector,
+        input_params: List[Dict] = [],
+        number_of_operands: int = 1,
+        warm_reset: bool = False,
+    ):
+        """Common verification function for all tests"""
+
+        input_source_flag: InputSourceFlags = None
+        if test_vector.input_source in (InputSource.FROM_DRAM_QUEUE,):
+            input_source_flag = InputSourceFlags.FROM_DRAM
+
+        operator = getattr(torch.nn, test_vector.operator)
+
+        kwargs = test_vector.kwargs if test_vector.kwargs else {}
+
+        model_type = cls.MODEL_TYPES[test_vector.input_source]
+        pytorch_model = model_type(
+            operator=operator,
+            opname=test_vector.operator,
+            shape=test_vector.input_shape,
+            kwargs=kwargs,
+        )
+
+        input_shapes = tuple([test_vector.input_shape for _ in range(number_of_operands)])
+        logger.trace(f"***input_shapes: {input_shapes}")
+
+        min = 0
+        max = test_vector.kwargs["num_embeddings"] - 1
+        match test_vector.input_source:
+            case InputSource.FROM_ANOTHER_OP:
+                max = int(max / 2)
+        value_range = ValueRange(min, max)
+
+        VerifyUtils.verify(
+            model=pytorch_model,
+            test_device=test_device,
+            input_shapes=input_shapes,
+            input_params=input_params,
+            input_source_flag=input_source_flag,
+            dev_data_format=test_vector.dev_data_format,
+            math_fidelity=test_vector.math_fidelity,
+            pcc=test_vector.pcc,
+            warm_reset=warm_reset,
+            value_range=value_range,
+        )
+
+
+class TestParamsData:
+
+    __test__ = False  # Avoid collecting TestParamsData as a pytest test
+
+    test_plan: TestPlan = None
+
+    rng = random.Random(31)
+    INPUT_SHAPE_THRESHOLD = 100000000
+    MAX_EMBEDDING_DIM = 10000
+
+    embedding_dims = [1000, 3200, MAX_EMBEDDING_DIM]
+    # weight_dtypes = [torch.bfloat16, torch.float32]
+
+    @classmethod
+    def generate_kwargs(cls, test_vector: TestVector, weight_dtype: Type[torch.dtype]):
```

Comment:
Currently test_single is not working because "parsing" of torch dtype is failing during kwargs eval.
Solution is to include `import torch` in `utils/plan.py`